BasicSteps/ProcessStep.cs

```diff
@@ -102,6 +132,11 @@ public override void Run()
             Int32 timeout = Timeout <= 0 ? Int32.MaxValue : Timeout;
             prepend = string.IsNullOrEmpty(LogHeader) ? "" : LogHeader + " ";
 
+            foreach (var environmentVariable in EnvironmentVariables)
+            {
+                Environment.SetEnvironmentVariable(environmentVariable.Name, environmentVariable.Value);
```

Comment:
Let's not set the environment variable for the parent process. Use ProcessStartInfo.EnvironmentVariables to set environment variables in the new process directly.

Reply:
Now it should set the environment variables only on the new process.